pages/deploy/deploy_aws.md

```diff
@@ -677,8 +679,39 @@ which will send any email bounces, delivery failures, rejects & complaints to an
 In order to do that go to AWS Console -> SNS -> Select the SNS topic which would look like `{resource_prefix}-{envname}-SNS-Email-Bounce-Topic` ( where resource_prefix and envname are specified in the cdk.json ) -> Create Subscription. You can attach multiple subscriptions to
 this SNS topic and monitor and take actions in case of any delivery failure.
 
+## Best practices and recommendations <a name="best"></a>
+
+### Deployment parameters
+Some of the deployment parameters in the `cdk.json` strenghten the security posture of the deployment. We encourage users
+to configure data.all with the following values.
+- Set `enable_pivot_role_auto_create` to `true`: it allows data.all to scope down permissions to the pivot role. 
+It also avoids manual management of pivot roles, which the subsequent reduction of manual errors.
+- Set `cognito_user_session_timeout_inmins` to the minimum: which constraints the impact of malicious actors impersonating a cognito user.
+
+
+### Least privilege permissions
+Data.all strives to fulfill this principle for all roles and personas using the platform. There are some additional
+guidelines that could serve customers to follow this principle when setting up AWS accounts to use data.all.
+
+- Access to the deployment account(s) should be restricted to data.all maintainer teams. IAM roles with limited permissions will be provided to these teams only.
+- Access to the tooling account should be restricted to developer teams. IAM roles with access only to the CICD 
+necessary resources will be provided to these teams only.
+- Access in the environment account(s). Data personas should use the provided IAM team roles to produce and consume data. Other IAM roles in the Environment 
+AWS Account should have limited permissions. You might use imported-IAM roles to data.all or consumption roles to adjust to your particular requirements.
+- Access in the environment account(s) for CDK execution role, should use the scoped-down CDK exec role policy that can 
+be downloaded from the data.all UI. This CloudFormation template can be used in the CDK bootstrap command.
+
+
+
+### Managing new releases and customizations
+
+To get the latest features and fixes, customers are encouraged to **keep in sync** with the latest version of data.all.
+At the same time, customers often develop their own features and customizations on top of data.all. We recommend 
+customers to **contribute back** these features so that we can manage them and respond to issues. Moreover, contributing back
+makes it easier to keep in sync with the latest data.all releases.
```

Comment:
I would add an extra line `"Please refer to the CONTRIBUTING.md file in the GitHub repository for more information on how to contribute back to data.all."`